django/contrib/admin/templatetags/admin_list.py

```diff
@@ -390,7 +413,15 @@ def date_hierarchy(cl):
                 } for month in months]
             }
         else:
-            years = getattr(cl.queryset, dates_or_datetimes)(field_name, 'year')
+            if approximate >= ApproximateWith.YEARS:
```

Comment:
Using `elif approximate >= ApproximateWith.YEARS:` and keeping the final `else` clause as is would remove a nesting level.